presidio-anonymizer/tests/test_aonnymizer_domain.py

```diff
@@ -0,0 +1,151 @@
+import json
+import os
+from typing import List
+
+import pytest
+
+from presidio_anonymizer.domain.analyzer_result import AnalyzerResult
+from presidio_anonymizer.domain.analyzer_results import AnalyzerResults
+from presidio_anonymizer.domain.anonymizer_request import AnonymizerRequest
+from presidio_anonymizer.domain.invalid_exception import InvalidJsonException
+
+
+@pytest.mark.parametrize(
+    "request_json, result_text",
+    [
+        ({}, "Invalid json, analyzer result must contain start",),
+        ({
+             "end": 32,
+             "score": 0.8,
+             "entity_type": "NUMBER"
+         }, "Invalid json, analyzer result must contain start",),
+        ({
+             "start": 28,
+             "score": 0.8,
+             "entity_type": "NUMBER"
+         }, "Invalid json, analyzer result must contain end",),
+        ({
+             "start": 28,
+             "end": 32,
+             "entity_type": "NUMBER"
+         }, "Invalid json, analyzer result must contain score",),
+        ({
+             "start": 28,
+             "end": 32,
+             "score": 0.8,
+         }, "Invalid json, analyzer result must contain entity_type",),
+    ],
+)
+def test_analyzer_result_invalid_json_formats(request_json, result_text):
+    try:
+        AnalyzerResult.validate_and_create(request_json)
+    except InvalidJsonException as e:
+        assert e.err == result_text
+    except Exception as e:
+        assert not e
+
+
+def test_analyzer_result_valid_json():
+    content = {
+        "start": 0,
+        "end": 32,
+        "score": 0.8,
+        "entity_type": "NUMBER"
+    }
+    data = AnalyzerResult.validate_and_create(content)
+    assert data.start == content.get("start")
+    assert data.end == content.get("end")
+    assert data.score == content.get("score")
+    assert data.entity_type == content.get("entity_type")
+
+
+@pytest.mark.parametrize(
+    "request_json, result_text",
+    [
```

Comment:
Black autoformats parametrized tests its to a non readable form, you can have it ignore those wrapping in a fmt: on/off block, see what I did in the flake8 plugins PR

Reply:
Done.

Reply:
Done.